### ISSUE TITLE
Inaccuracy in TLS configuration in user guide for stanza section.

### ISSUE THREAD

**Issue by woblerr** [pgbackrest/pgbackrest#1644]
1. pgBackRest version: 2.37

2. PostgreSQL version: 13

3. Operating system: Ubuntu 20.04

4. Did you install pgBackRest from source or from a package? Package

5. Describe the issue:

[In the dedicated repository host documentation ](https://pgbackrest.org/user-guide-rhel.html#repo-host) (paragraph 16.2) for the `pg-primary` host configuration, some parameters are located in the stanza section (`repo1-host-ca-file, repo1-host-cert-file, repo1-host-key-file, repo1-host-type`).

However, with this configuration, on `pg-primary` host pgBackRest tries to connect via ssh without using a TLS connection.

<details>
<summary>pgbackrest.conf (pg-primary):</summary>

```bash
[demo]
pg1-path=/var/lib/postgresql/13/main
repo1-host-ca-file=/etc/pgbackrest/cert/ca.crt
repo1-host-cert-file=/etc/pgbackrest/cert/client.crt
repo1-host-key-file=/etc/pgbackrest/cert/client.key
repo1-host-type=tls

[global]
repo1-host=backup
repo1-retention-diff=2
repo1-retention-full=2
start-fast=y
log-level-console=info
tls-server-port=8432
tls-server-address=*
tls-server-auth=pgbackrest-client=*
tls-server-ca-file=/etc/pgbackrest/cert/pgbackrest-test-ca.crt
tls-server-cert-file=/etc/pgbackrest/cert/pgbackrest-test-server.crt
tls-server-key-file=/etc/pgbackrest/cert/pgbackrest-test-server.key

[global:archive-push]
compress-level=3
```

</details>

<details>
<summary>pgbackrest info --log-level-console=debug :</summary>

```bash
2022-01-31 23:08:43.832 P00  DEBUG:     common/io/socket/common::sckInit: (block: false, keepAlive: true, tcpKeepAliveCount: 0, tcpKeepAliveIdle: 0, tcpKeepAliveInterval: 0)
2022-01-31 23:08:43.835 P00  DEBUG:     common/io/socket/common::sckInit: => void
2022-01-31 23:08:43.835 P00  DEBUG:     command/command::cmdBegin: info command begin 2.37: --exec-id=14966-1598820e --log-level-console=debug --repo1-host=backup
2022-01-31 23:08:43.848 P00  DEBUG:     config/load::cfgLoad: => void
2022-01-31 23:08:43.848 P00  DEBUG:     command/info/info::cmdInfo: (void)
2022-01-31 23:08:43.848 P00  DEBUG:     command/info/info::infoRender: (void)
2022-01-31 23:08:43.848 P00  DEBUG:     protocol/helper::repoIsLocal: (repoIdx: 0)
2022-01-31 23:08:43.848 P00  DEBUG:     protocol/helper::repoIsLocal: => false
2022-01-31 23:08:43.848 P00  DEBUG:     protocol/helper::protocolRemoteGet: (protocolStorageType: repo, hostIdx: 0)
2022-01-31 23:08:43.848 P00  DEBUG:     protocol/helper::protocolRemoteParamSsh: (protocolStorageType: repo, hostIdx: 0)
2022-01-31 23:08:43.848 P00  DEBUG:     protocol/helper::protocolRemoteParam: (protocolStorageType: repo, hostIdx: 0)
2022-01-31 23:08:43.848 P00  DEBUG:     protocol/helper::protocolRemoteParam: => {["--exec-id=14966-1598820e", "--log-level-console=off", "--log-level-file=off", "--log-level-stderr=error", "--process=0", "--remote-type=repo", "--repo=1", "info:remote"]}
2022-01-31 23:08:43.848 P00  DEBUG:     protocol/helper::protocolRemoteParamSsh: => {["-o", "LogLevel=error", "-o", "Compression=no", "-o", "PasswordAuthentication=no", "pgbackrest@backup", "pgbackrest --exec-id=14966-1598820e --log-level-console=off --log-level-file=off --log-level-stderr=error --process=0 --remote-type=repo --repo=1 info:remote"]}
2022-01-31 23:08:43.848 P00  DEBUG:     common/exec::execNew: (command: {"ssh"}, param: {["-o", "LogLevel=error", "-o", "Compression=no", "-o", "PasswordAuthentication=no", "pgbackrest@backup", "pgbackrest --exec-id=14966-1598820e --log-level-console=off --log-level-file=off --log-level-stderr=error --process=0 --remote-type=repo --repo=1 info:remote"]}, name: {"remote-0 process on 'backup'"}, timeout: 1830000)
2022-01-31 23:08:43.848 P00  DEBUG:     common/exec::execNew: => {Exec}
2022-01-31 23:08:43.848 P00  DEBUG:     common/exec::execOpen: (this: {Exec})
2022-01-31 23:08:43.849 P00  DEBUG:     common/exec::execOpen: => void
2022-01-31 23:08:43.956 P00  DEBUG:     command/info/info::infoRender: => {"stanza: [invalid]
                                            status: error (other)
                                                    [UnknownError] remote-0 process on 'backup' terminated unexpectedly [255]: pgbackrest@backup: Permission denied (publickey).
                                            cipher: none
                                        "}
stanza: [invalid]
    status: error (other)
            [UnknownError] remote-0 process on 'backup' terminated unexpectedly [255]: pgbackrest@backup: Permission denied (publickey).
    cipher: none
2022-01-31 23:08:43.960 P00  DEBUG:     command/info/info::cmdInfo: => void
2022-01-31 23:08:43.961 P00  DEBUG:     common/exit::exitSafe: (result: 0, error: false, signalType: 0)
2022-01-31 23:08:43.961 P00  DEBUG:     command/command::cmdEnd: info command end: completed successfully (130ms)
2022-01-31 23:08:43.961 P00  DEBUG:     common/lock::lockRelease: (failOnNoLock: false)
2022-01-31 23:08:43.961 P00  DEBUG:     common/lock::lockRelease: => false
2022-01-31 23:08:43.961 P00  DEBUG:     common/exit::exitSafe: => 0
2022-01-31 23:08:43.961 P00  DEBUG:     main::main: => 0
```

</details>

If the parameters are moved to the `[global]` section, pgBackRest will start using a TLS connection.

<details>
<summary>pgbackrest.conf (pg-primary):</summary>

```bash
[demo]
pg1-path=/var/lib/postgresql/13/main

[global]
repo1-host=backup
repo1-host-ca-file=/etc/pgbackrest/cert/pgbackrest-test-ca.crt
repo1-host-cert-file=/etc/pgbackrest/cert/pgbackrest-test-client.crt
repo1-host-key-file=/etc/pgbackrest/cert/pgbackrest-test-client.key
repo1-host-type=tls
repo1-retention-diff=2
repo1-retention-full=2
start-fast=y
log-level-console=info
tls-server-port=8432
tls-server-address=*
tls-server-auth=pgbackrest-client=*
tls-server-ca-file=/etc/pgbackrest/cert/pgbackrest-test-ca.crt
tls-server-cert-file=/etc/pgbackrest/cert/pgbackrest-test-server.crt
tls-server-key-file=/etc/pgbackrest/cert/pgbackrest-test-server.key

[global:archive-push]
compress-level=3
```

</details>

<details>
<summary>pgbackrest info --log-level-console=debug :</summary>

```bash
2022-01-31 23:15:32.204 P00  DEBUG:     common/io/socket/common::sckInit: (block: false, keepAlive: true, tcpKeepAliveCount: 0, tcpKeepAliveIdle: 0, tcpKeepAliveInterval: 0)
2022-01-31 23:15:32.205 P00  DEBUG:     common/io/socket/common::sckInit: => void
2022-01-31 23:15:32.205 P00  DEBUG:     command/command::cmdBegin: info command begin 2.37: --exec-id=15000-69c2a949 --log-level-console=debug --repo1-host=backup --repo1-host-ca-file=/etc/pgbackrest/cert/pgbackrest-test-ca.crt --repo1-host-cert-file=/etc/pgbackrest/cert/pgbackrest-test-client.crt --repo1-host-key-file=/etc/pgbackrest/cert/pgbackrest-test-client.key --repo1-host-type=tls
2022-01-31 23:15:32.205 P00  DEBUG:     config/load::cfgLoad: => void
2022-01-31 23:15:32.205 P00  DEBUG:     command/info/info::cmdInfo: (void)
2022-01-31 23:15:32.206 P00  DEBUG:     command/info/info::infoRender: (void)
2022-01-31 23:15:32.206 P00  DEBUG:     protocol/helper::repoIsLocal: (repoIdx: 0)
2022-01-31 23:15:32.206 P00  DEBUG:     protocol/helper::repoIsLocal: => false
2022-01-31 23:15:32.206 P00  DEBUG:     protocol/helper::protocolRemoteGet: (protocolStorageType: repo, hostIdx: 0)
2022-01-31 23:15:32.206 P00  DEBUG:     common/io/socket/client::sckClientNew: (host: {"backup"}, port: 8432, timeoutConnect: 60000, timeoutSession: 1830000)
2022-01-31 23:15:32.206 P00  DEBUG:     common/io/socket/client::sckClientNew: => {type: socket, driver: {host: backup, port: 8432, timeoutConnect: 60000, timeoutSession: 1830000}}
2022-01-31 23:15:32.206 P00  DEBUG:     common/io/tls/client::tlsClientNew: (ioClient: {type: socket, driver: {host: backup, port: 8432, timeoutConnect: 60000, timeoutSession: 1830000}}, host: {"backup"}, timeoutConnect: 60000, timeoutSession: 1830000, verifyPeer: true, param.caFile: {"/etc/pgbackrest/cert/pgbackrest-test-ca.crt"}, param.caPath: null, param.certFile: {"/etc/pgbackrest/cert/pgbackrest-test-client.crt"}, param.keyFile: {"/etc/pgbackrest/cert/pgbackrest-test-client.key"})
2022-01-31 23:15:32.209 P00  DEBUG:     storage/posix/storage::storagePosixNew: (path: {"/"}, param.modeFile: 0000, param.modePath: 0000, param.write: false, param.pathExpressionFunction: null)
2022-01-31 23:15:32.209 P00  DEBUG:     storage/posix/storage::storagePosixNewInternal: (type: posix, path: {"/"}, modeFile: 0640, modePath: 0750, write: false, pathExpressionFunction: null, pathSync: true)
2022-01-31 23:15:32.209 P00  DEBUG:     storage/posix/storage::storagePosixNewInternal: => {type: posix, path: {"/"}, write: false}
2022-01-31 23:15:32.209 P00  DEBUG:     storage/posix/storage::storagePosixNew: => {type: posix, path: {"/"}, write: false}
2022-01-31 23:15:32.209 P00  DEBUG:     storage/storage::storageInfo: (this: {type: posix, path: {"/"}, write: false}, fileExp: {"/etc/pgbackrest/cert/pgbackrest-test-client.key"}, param.level: 0, param.ignoreMissing: true, param.followLink: true, param.noPathEnforce: false)
2022-01-31 23:15:32.210 P00  DEBUG:     storage/storage::storageInfo: => {StorageInfo}
2022-01-31 23:15:32.210 P00  DEBUG:     common/io/tls/client::tlsClientNew: => {type: tls, driver: {ioClient: {type: socket, driver: {host: backup, port: 8432, timeoutConnect: 60000, timeoutSession: 1830000}}, timeoutConnect: 60000, timeoutSession: 1830000, verifyPeer: true}}
2022-01-31 23:15:32.254 P00  DEBUG:     protocol/helper::protocolRemoteParam: (protocolStorageType: repo, hostIdx: 0)
2022-01-31 23:15:32.254 P00  DEBUG:     protocol/helper::protocolRemoteParam: => {["--exec-id=15000-69c2a949", "--log-level-console=off", "--log-level-file=off", "--log-level-stderr=error", "--process=0", "--remote-type=repo", "--repo=1", "info:remote"]}
2022-01-31 23:15:32.256 P00  DEBUG:     protocol/helper::protocolRemoteGet: => {name: remote-0 tls protocol on 'backup', state: idle}
2022-01-31 23:15:32.256 P00  DEBUG:     storage/remote/storage::storageRemoteNew: (modeFile: 0640, modePath: 0750, write: false, pathExpressionFunction: (function *), client: {name: remote-0 tls protocol on 'backup', state: idle}, compressLevel: 3)
2022-01-31 23:15:32.256 P00  DEBUG:     storage/remote/storage::storageRemoteNew: => {type: remote, path: {"/var/lib/pgbackrest"}, write: false}
2022-01-31 23:15:32.256 P00  DEBUG:     storage/storage::storageList: (this: {type: remote, path: {"/var/lib/pgbackrest"}, write: false}, pathExp: {"backup"}, param.errorOnMissing: false, param.nullOnMissing: false, param.expression: null)
2022-01-31 23:15:32.256 P00  DEBUG:     storage/storage::storageInfoList: (this: {type: remote, path: {"/var/lib/pgbackrest"}, write: false}, pathExp: {"backup"}, callback: (function *), callbackData: *void, param.level: 1, param.errorOnMissing: false, param.sortOrder: 0, param.expression: null, param.recurse: false)
2022-01-31 23:15:32.257 P00  DEBUG:     storage/storage::storageInfoList: => true
2022-01-31 23:15:32.257 P00  DEBUG:     storage/storage::storageList: => {["demo"]}
2022-01-31 23:15:32.257 P00  DEBUG:     info/infoBackup::infoBackupLoadFile: (storage: {type: remote, path: {"/var/lib/pgbackrest"}, write: false}, fileName: {"backup/demo/backup.info"}, cipherType: none)
2022-01-31 23:15:32.258 P00  DEBUG:     info/infoBackup::infoBackupLoadFile: => {InfoBackup}
2022-01-31 23:15:32.258 P00  DEBUG:     info/infoArchive::infoArchiveLoadFile: (storage: {type: remote, path: {"/var/lib/pgbackrest"}, write: false}, fileName: {"archive/demo/archive.info"}, cipherType: none)
2022-01-31 23:15:32.259 P00  DEBUG:     info/infoArchive::infoArchiveLoadFile: => {InfoArchive}
2022-01-31 23:15:32.259 P00  DEBUG:     common/lock::lockAcquire: (lockPath: {"/tmp/pgbackrest"}, stanza: {"demo"}, execId: {"15000-69c2a949"}, lockType: 1, lockTimeout: 0, failOnNoLock: false)
2022-01-31 23:15:32.259 P00  DEBUG:     common/lock::lockAcquire: => true
2022-01-31 23:15:32.259 P00  DEBUG:     common/lock::lockRelease: (failOnNoLock: true)
2022-01-31 23:15:32.259 P00  DEBUG:     common/lock::lockRelease: => true
2022-01-31 23:15:32.259 P00  DEBUG:     info/infoPg::infoPgDataCurrentId: (this: {InfoPg})
2022-01-31 23:15:32.260 P00  DEBUG:     info/infoPg::infoPgDataCurrentId: => 0
2022-01-31 23:15:32.260 P00  DEBUG:     storage/storage::storageList: (this: {type: remote, path: {"/var/lib/pgbackrest"}, write: false}, pathExp: {"archive/demo/13-1"}, param.errorOnMissing: false, param.nullOnMissing: false, param.expression: {"^[0-F]{16}$"})
2022-01-31 23:15:32.260 P00  DEBUG:     storage/storage::storageInfoList: (this: {type: remote, path: {"/var/lib/pgbackrest"}, write: false}, pathExp: {"archive/demo/13-1"}, callback: (function *), callbackData: *void, param.level: 1, param.errorOnMissing: false, param.sortOrder: 0, param.expression: {"^[0-F]{16}$"}, param.recurse: false)
2022-01-31 23:15:32.260 P00  DEBUG:     storage/storage::storageInfoList: => true
2022-01-31 23:15:32.260 P00  DEBUG:     storage/storage::storageList: => {["0000000100000000"]}
2022-01-31 23:15:32.260 P00  DEBUG:     storage/storage::storageList: (this: {type: remote, path: {"/var/lib/pgbackrest"}, write: false}, pathExp: {"archive/demo/13-1/0000000100000000"}, param.errorOnMissing: false, param.nullOnMissing: false, param.expression: {"^[0-F]{24}-[0-f]{40}(\.gz|\.lz4|\.zst|\.xz|\.bz2){0,1}$"})
2022-01-31 23:15:32.260 P00  DEBUG:     storage/storage::storageInfoList: (this: {type: remote, path: {"/var/lib/pgbackrest"}, write: false}, pathExp: {"archive/demo/13-1/0000000100000000"}, callback: (function *), callbackData: *void, param.level: 1, param.errorOnMissing: false, param.sortOrder: 0, param.expression: {"^[0-F]{24}-[0-f]{40}(\.gz|\.lz4|\.zst|\.xz|\.bz2){0,1}$"}, param.recurse: false)
2022-01-31 23:15:32.262 P00  DEBUG:     storage/storage::storageInfoList: => true
2022-01-31 23:15:32.262 P00  DEBUG:     storage/storage::storageList: => {["000000010000000000000006-684f5d28748380dbdc86669935ff615aa63326af.gz", "000000010000000000000007-1dcdb898185d8a45aa5522656e3e661a77847a82.gz", "00000001000000000000000A-cface13bd39b8fb0fcd010fb96ae9168e187fc09.gz", "000000010000000000000008-cea8552ebc41ad158eb194062067351032381c74.gz", "000000010000000000000004-4aae2175c0821034f5ed0c87487b8e6df8a2d521.gz", "00000001000000000000000B-6fc67b77d206cdcc56c24fa1a104f7c71283a0aa.gz", "000000010000000000000005-72e2d539a56de23138754eba93643f2b9c26312a.gz", "000000010000000000000009-6dafc0ac6d7154f118053422a8a1799934dabc74.gz"]}
2022-01-31 23:15:32.262 P00  DEBUG:     storage/storage::storageList: (this: {type: remote, path: {"/var/lib/pgbackrest"}, write: false}, pathExp: {"archive/demo/13-1/0000000100000000"}, param.errorOnMissing: false, param.nullOnMissing: false, param.expression: {"^[0-F]{24}-[0-f]{40}(\.gz|\.lz4|\.zst|\.xz|\.bz2){0,1}$"})
2022-01-31 23:15:32.262 P00  DEBUG:     storage/storage::storageInfoList: (this: {type: remote, path: {"/var/lib/pgbackrest"}, write: false}, pathExp: {"archive/demo/13-1/0000000100000000"}, callback: (function *), callbackData: *void, param.level: 1, param.errorOnMissing: false, param.sortOrder: 0, param.expression: {"^[0-F]{24}-[0-f]{40}(\.gz|\.lz4|\.zst|\.xz|\.bz2){0,1}$"}, param.recurse: false)
2022-01-31 23:15:32.263 P00  DEBUG:     storage/storage::storageInfoList: => true
2022-01-31 23:15:32.263 P00  DEBUG:     storage/storage::storageList: => {["000000010000000000000006-684f5d28748380dbdc86669935ff615aa63326af.gz", "000000010000000000000007-1dcdb898185d8a45aa5522656e3e661a77847a82.gz", "00000001000000000000000A-cface13bd39b8fb0fcd010fb96ae9168e187fc09.gz", "000000010000000000000008-cea8552ebc41ad158eb194062067351032381c74.gz", "000000010000000000000004-4aae2175c0821034f5ed0c87487b8e6df8a2d521.gz", "00000001000000000000000B-6fc67b77d206cdcc56c24fa1a104f7c71283a0aa.gz", "000000010000000000000005-72e2d539a56de23138754eba93643f2b9c26312a.gz", "000000010000000000000009-6dafc0ac6d7154f118053422a8a1799934dabc74.gz"]}
2022-01-31 23:15:32.263 P00  DEBUG:     info/infoPg::infoPgDataCurrentId: (this: {InfoPg})
2022-01-31 23:15:32.263 P00  DEBUG:     info/infoPg::infoPgDataCurrentId: => 0
2022-01-31 23:15:32.264 P00  DEBUG:     command/info/info::infoRender: => {"stanza: demo
                                            status: ok
                                            cipher: none
                                        
                                            db (current)
                                                wal archive min/max (13): 000000010000000000000004/00000001000000000000000B
                                        
                                                full backup: 20220201-005022F
                                                    timestamp start/stop: 2022-01-31 21:50:22 / 2022-01-31 21:50:27
                                                    wal start/stop: 000000010000000000000007 / 000000010000000000000007
                                                    database size: 23.2MB, database backup size: 23.2MB
                                                    repo1: backup set size: 2.8MB, backup size: 2.8MB
                                        "}
stanza: demo
    status: ok
    cipher: none

    db (current)
        wal archive min/max (13): 000000010000000000000004/00000001000000000000000B

        full backup: 20220201-005022F
            timestamp start/stop: 2022-01-31 21:50:22 / 2022-01-31 21:50:27
            wal start/stop: 000000010000000000000007 / 000000010000000000000007
            database size: 23.2MB, database backup size: 23.2MB
            repo1: backup set size: 2.8MB, backup size: 2.8MB
2022-01-31 23:15:32.266 P00  DEBUG:     command/info/info::cmdInfo: => void
2022-01-31 23:15:32.266 P00  DEBUG:     common/exit::exitSafe: (result: 0, error: false, signalType: 0)
2022-01-31 23:15:32.267 P00 DETAIL: statistics: {"socket.client":{"total":1},"socket.session":{"total":1},"tls.client":{"total":1},"tls.session":{"total":1}}
2022-01-31 23:15:32.267 P00  DEBUG:     command/command::cmdEnd: info command end: completed successfully (64ms)
2022-01-31 23:15:32.267 P00  DEBUG:     common/lock::lockRelease: (failOnNoLock: false)
2022-01-31 23:15:32.267 P00  DEBUG:     common/lock::lockRelease: => false
2022-01-31 23:15:32.267 P00  DEBUG:     common/exit::exitSafe: => 0
2022-01-31 23:15:32.267 P00  DEBUG:     main::main: => 0
```

</details>

It looks like an inaccuracy in the documentation or a problem with parsing parameters.

**Comment by dwsteele** [comment]
Yeah, these definitely need to be in the `global` section for info to work correctly. Other commands are fine since they specify a stanza. We'll fix this and update the docs.

**Comment by dwsteele** [comment]
Fixed at 53de3e3aeb71f8209f56167941df35bbfcb7cdc9. Since the release is close we'll wait to update the online docs until then,
